src/Core/Managed/Shared/DataContracts/AggregatedMetricTelemetry.cs

```diff
@@ -0,0 +1,157 @@
+
+        internal readonly string BaseType = typeof(MetricData).Name;
+
+        internal readonly MetricData Data;
+        internal readonly DataPoint Metric;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AggregatedMetricTelemetry"/> class with empty 
+        /// properties.
+        /// </summary>
+        public AggregatedMetricTelemetry()
+        {
+            this.Data = new MetricData();
+            this.Metric = new DataPoint();
+            this.Context = new TelemetryContext(this.Data.properties);
+
+            // We always have a single 'metric'.
+            this.Data.metrics.Add(this.Metric);
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AggregatedMetricTelemetry"/> class with properties provided.
+        /// </summary>
+        /// <remarks>
+        /// Metric statistics provided are assumed to be calculated over a period of time equaling 1 minute.
+        /// </remarks>
+        /// <param name="metricName">Metric name.</param>
+        /// <param name="count">Count of values taken during aggregation interval.</param>
+        /// <param name="sum">Sum of values taken during aggregation interval.</param>
+        /// <param name="min">Minimum value taken during aggregation interval.</param>
+        /// <param name="max">Maximum of values taken during aggregation interval.</param>
+        /// <param name="standardDeviation">Standard deviation of values taken during aggregation interval.</param>
+        public AggregatedMetricTelemetry(
+            string metricName, 
+            int count,
+            double sum,
+            double min,
+            double max,
+            double standardDeviation)
+            : this()
+        {
+            this.Name = metricName;
+            this.Count = count;
+            this.Sum = sum;
+            this.Min = min;
+            this.Max = max;
+            this.StandardDeviation = standardDeviation;
+        }
+
+        /// <summary>
+        /// Gets or sets date and time when event was recorded.
+        /// </summary>
+        public DateTimeOffset Timestamp { get; set; }
+
+        /// <summary>
+        /// Gets or sets the value that defines absolute order of the telemetry item.
+        /// </summary>
+        public string Sequence { get; set; }
+
+        /// <summary>
+        /// Gets the context associated with the current telemetry item.
+        /// </summary>
+        public TelemetryContext Context { get; }
+
+        /// <summary>
+        /// Gets or sets the name of the metric.
+        /// </summary>
+        public string Name
+        {
+            get { return this.Metric.name; }
+            set { this.Metric.name = value; }
+        }
+
+        /// <summary>
+        /// Gets or sets the number of samples for this metric.
+        /// </summary>
+        public int Count
+        {
+            get { return this.Metric.count ?? 0; }
```

Comment:
Sanitize will set Count to 1 only in case when AggregatedMetricTelemetry was constructed w/o setting "Count" property (or explicitly setting it to <= 0). We consider this a mistake and I initially was setting it to "0" but that may present some problems in the pipeline...

The statement about different value on the wire and through property getter is not unique to this particular case.